### PR TITLE
feat(Core/Unit): Implement munching as a configurable option.

### PR DESCRIPTION
**Pull request by heyitsbench**
## Changes Proposed:

This PR proposes changes to:
-  [X] Core (units).

## Issues Addressed:

- Closes https://github.com/azerothcore/azerothcore-wotlk/issues/18593
- Progresses https://github.com/chromiecraft/chromiecraft/issues/5669
- Progresses https://github.com/chromiecraft/chromiecraft/issues/7028

## SOURCE:

This config creates a deviation from Blizzlike under the custom category of the worldserver.conf. The change has been made on many private servers and Blizzard's own Wrath Classic as a quality of life change.

## Tests Performed:

This PR:
- [X] Has not been tested.

## How to Test the Changes:

- [X] This pull request can be tested by following the reproduction steps provided in the linked issue.

## How to Test AzerothCore PRs
 
When a PR is ready to be tested, it will be marked as **[WAITING TO BE TESTED]**.

You can help by testing PRs and writing your feedback here on the PR's page on GitHub. Follow the instructions here:

http://www.azerothcore.org/wiki/How-to-test-a-PR

**REMEMBER**: when testing a PR that changes something **generic** (i.e. a part of code that handles more than one specific thing), the tester should not only check that the PR does its job (e.g. fixing spell XXX) but **especially** check that the PR does not cause any regression (i.e. introducing new bugs).

**For example**: if a PR fixes spell X by changing a part of code that handles spells X, Y, and Z, we should not only test X, but **we should test Y and Z as well**.

### CODE REVIEW

**Comment by heyitsbench** [comment]
It has been a bit since I've added any configs, so forgive me if I made a mistake.

File: src/server/apps/worldserver/worldserver.conf.dist [hunk 1]
```diff
@@ -4379,6 +4379,13 @@ TeleportTimeoutFar = 45
 
 DailyRBGArenaPoints.MinLevel = 71
 
+#
+#    MunchingBlizzlike.Enabled
+#        Description: Disable the Blizzlike implementation of munching with e.g. Warrior's Rend or Mage's Ignite
```
[the remark this answers is not in the document]
Comment (heyitsbench) [reply src/server/apps/worldserver/worldserver.conf.dist]: Corrected.

File: src/server/game/World/World.cpp [hunk 2]
```diff
@@ -1282,6 +1282,8 @@ void World::LoadConfigSettings(bool reload)
 
     _bool_configs[CONFIG_ALLOWS_RANK_MOD_FOR_PET_HEALTH] = sConfigMgr->GetOption<bool>("Pet.RankMod.Health", true);
 
+    _bool_configs[CONFIG_MUNCHING_BLIZZLIKE] = sConfigMgr->GetOption<bool>("MunchingBlizzlike.Enabled", true);
```
Comment (Kitzunu) [inline src/server/game/World/World.cpp:1285]: As you only use this in one place you could technically move `sConfigMgr->GetOption<bool>("MunchingBlizzlike.Enabled", true)` directly into the if statement and not have it in world files. But it doesn't matter

Comment (heyitsbench) [reply src/server/game/World/World.cpp:1285]: Personally I prefer how it is currently for consistency, but that's just me.

**Comment by avarishd** [comment]
May want to change the config to 0 and 1, as everything else follows that.

**Review by Nyeriah — APPROVED** [review]

File: src/server/game/World/World.cpp (hunk 2 above)
Comment (Nyeriah) [reply src/server/game/World/World.cpp:1285]: My understanding is that every config should be loaded on start up to save up on I/O operations from reading the text file, which are rather resource intensive

File: src/server/apps/worldserver/worldserver.conf.dist [hunk 3]
```diff
@@ -4379,6 +4379,13 @@ TeleportTimeoutFar = 45
 
 DailyRBGArenaPoints.MinLevel = 71
 
+#
+#    MunchingBlizzlike.Enabled
+#        Description: Enable the Blizzlike implementation of munching with e.g. Warrior's Rend or Mage's Ignite
+#        Default: true - (Blizzlike)
+
+MunchingBlizzlike.Enabled = true
```
Comment (Nyeriah) [inline src/server/apps/worldserver/worldserver.conf.dist:4387]:
```suggestion
MunchingBlizzlike.Enabled = 1
```

**Comment by heyitsbench** [comment]
> May want to change the config to 0 and 1, as everything else follows that.

As much as I don't like it, done.

**Pull request merged**